package/gluon-xlat464-clat/files/xlat464clat.sh

```diff
@@ -0,0 +1,173 @@
+# GNU General Public License for more details.
+
+CTR=/proc/net/nat46/control
+
+
+[ -n "$INCLUDE_ONLY" ] || {
+  . /lib/functions.sh
+  . ../netifd-proto.sh
+  init_proto "$@"
+}
+
+
+proto_xlat464clat_init_config() {
+  proto_config_add_string "zone"
+  proto_config_add_string "zone4"
+  proto_config_add_string "local_style"
+  proto_config_add_string "local_v4"
+  proto_config_add_string "local_v6"
+  proto_config_add_string "local_ea_len"
+  proto_config_add_string "local_psid"
+  proto_config_add_boolean "local_fmr_flag"
+  proto_config_add_string "remote_style"
+  proto_config_add_string "remote_v4"
+  proto_config_add_string "remote_v6"
+  proto_config_add_string "remote_ea_len"
+  proto_config_add_string "remote_psid"
+  proto_config_add_string "ip4table"
+  proto_config_add_boolean "debug"
+  available=1
+  no_device=1
+}
+
+
+proto_xlat464clat_setup() {
+  local config="$1"
+  local clat_cfg="config ${config}"
+  local local_style
+  local local_v4
+  local local_v6
+  local local_ea_len
+  local local_psid
+  local local_fmr
+  local remote_style
+  local remote_v4
+  local remote_v6
+  local remote_ea_len
+  local remote_psid
+  local zone
+  local zone4
+  local ip4table
+
+
+  config_load network
+
+  config_get		ip4table	"${config}" "ip4table"
+  config_get		zone		"${config}" "zone"
+  config_get		zone4		"${config}" "zone4"
+  config_get_bool	debug		"${config}" "debug" 0
+  config_get		local_style	"${config}" "local_style"
+  config_get		local_v4	"${config}" "local_v4"
+  config_get		local_v6	"${config}" "local_v6"
+  config_get		local_ea_len	"${config}" "local_ea_len"
+  config_get		local_psid	"${config}" "local_psid_offset"
+  config_get_bool	local_fmr	"${config}" "local_fmr_flag" 0
+  config_get		remote_style	"${config}" "remote_style"
+  config_get		remote_v4	"${config}" "remote_v4"
+  config_get		remote_v6	"${config}" "remote_v6"
+  config_get		remote_ea_len	"${config}" "remote_ea_len"
+  config_get		remote_psid	"${config}" "remote_psid_offset"
+
+
+  zone="${zone:-wan}"
+  zone4="${zone4:-lan}"
+  local_v6="${local_v6:-fd00:13:37:13:37::/96}"
+  remote_v4="${remote_v4:-0.0.0.0/0}"
+
+  [ $debug -ne 0 ] && clat_cfg="$clat_cfg debug 1"
+  [ ${local_fmr} -ne 0 ] && clat_cfg="$clat_cfg local.fmr-flag 1"
+
+  clat_cfg="${clat_cfg} local.style ${local_style:-RFC6052} local.v4 ${local_v4:-192.168.1.0/24} local.v6 ${local_v6}"
+  clat_cfg="${clat_cfg} local.ea-len ${local_ea_len:-0} local.psid-offset ${local_psid:-0}"
+  clat_cfg="${clat_cfg} remote.style ${remote_style:-RFC6052} remote.v4 ${remote_v4} remote.v6 ${remote_v6:-64:ff9b::/96}"
+  clat_cfg="${clat_cfg} remote.ea-len ${remote_ea_len:-0} remote.psid-offset ${remote_psid:-0}"
+
+  # create nat46 interface
+  echo "add ${config}" > ${CTR}
+
+  # configure nat46
+  echo "${clat_cfg}" > ${CTR}
+
+  [ "${ip4table}" ] && ip -4 rule add from ${local_v4} lookup ${ip4table}
+
+  proto_init_update "${config}" 1
+
+	  # add routes
```

Comment:
suddenly the indentation changes? please check the complete file to have an indentation matching other shell scripts in Gluon